src/ceph-daemon

```diff
@@ -809,6 +809,10 @@ def command_version():
 ##################################
 
 def command_bootstrap():
+    # some initial validation
+    if not args.skip_dashboard and not args.initial_dashboard_password:
+        raise RuntimeError('Must specify either --initial-dashboard-password <password> or --skip-dashboard')
+
     fsid = args.fsid or make_fsid()
     hostname = get_hostname()
```

Comment:
I just looked at the hostname function and shouldn't we call socket.getfqdn() first and only if the fqdn isn't set we could query socket.gethostname() ?

Reply:
for the dashboard URL, yes; for the mon name, i think we want the short name. will fix.